connections/ethereum/gsn/gsn.go

```diff
@@ -0,0 +1,81 @@
+package gsn
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"time"
+)
+
+type RPCError struct {
+	Code    int         `json:"code"`
+	Message string      `json:"message"`
+	Data    interface{} `json:"data,omitempty"`
+}
+
+type Response struct {
+	Error  *RPCError       `json:"error"`
+	ID     int             `json:"id"`
+	Result json.RawMessage `json:"result,omitempty"`
+}
+
+type gsnResponse struct {
+	Fast          int64       `json:"fast"`
+	Fastest       int64       `json:"fastest"`
+	SafeLow       int64       `json:"safeLow"`
+	Average       int64       `json:"average"`
+	Block_time    float32     `json:"block_time"`
+	BlockNum      int64       `json:"blockNum"`
+	Speed         float32     `json:"speed"`
+	SafeLowWait   float32     `json:"safeLowWait"`
+	AvgWait       float32     `json:"avgWait"`
+	FastWait      float32     `json:"fastWait"`
+	FastestWait   float32     `json:"fastestWait"`
+	GasPriceRange interface{} `json:"gasPriceRange"`
+}
+
+// callGSN will call the Gas Station Network and request the gas prices on the Ethereum network
+func CallGSN(apiKey, speed string) (error, *big.Int) {
+	time.Sleep(1 * time.Second)
+
+	var gsnURL = "https://ethgasstation.info/api/ethgasAPI.json?api-key=" + apiKey
+
+	var gsnRes gsnResponse
+	var body []byte
+
+	retries := 10
+	for i := 0; i < retries; i++ {
+		res, err := http.Get(gsnURL)
```

Comment:
```suggestion
		res, err := http.Get(filepath.Clean(gsnURL))
```

Reply:
This doesn't solve things completely. Basically, using the default client as we are here means that there is no timeout for requests, which could cause the code to block indefinitely. Afaik that best way to fix this is something like:
```
client := &http.Client{Timeout: time.Second * 30}
client.Get(...)
```

Reply:
Also, we need to call `res.Body.Close()`. You can use `defer res.Body.Close()` to ensure it will be called if an error is thrown below.